### ISSUE TITLE
[Select] Discrepancy in component behavior

### ISSUE THREAD

**Issue by Kritvi-bhatia17** [carbon-design-system/carbon-design-kit#908]
## Acceptance criteria
The Select component should no longer display placeholder text such as "Option." In both Figma and Code, the component either shows an empty option selected by default or displays one of the named options from the list. Once a user selects a different option, the empty option should be deselected. It should not fall back to displaying "Option" or any placeholder text.

### Tasks
- [ ] Match the Select component behavior as shown in [Storybook](https://react.carbondesignsystem.com/?path=/story/components-select--playground): by default, the selected option should be empty. When a new option is chosen, the empty option should be deselected, and the newly selected option should appear—without reverting to a placeholder like "Option".
- [ ] Review 

## Additional information
![image](https://github.com/carbon-design-system/carbon/assets/158570656/cb1db461-6c22-424e-9b27-c06f9db9b280)

**Comment by tay1orjones** [comment]
The option with the lengthy text string is so that we can validate the truncation ellipsis works as expected

![Image](https://github.com/carbon-design-system/carbon-design-kit/assets/3360588/b08686b2-42fc-4bdc-b5e1-85d7da6b2e18)

We could change this to be lower in the option list, but arguably this long string example should be present in figma to showcase the truncation ellipsis functionality to match storybook.

**Comment by Kritvi-bhatia17** [comment]
Yes, the long text is fine, and it's a smart idea to show the same in the Figma kit as well. 

However, regarding the issue I mentioned: when the first option is deselected, the text should appear. 
Let me confirm internally with the team if we want to simplify the interaction by always displaying the placeholder value.

https://github.com/carbon-design-system/carbon-design-kit/assets/158570656/09019f09-ddd6-4e28-addd-4d5cdad465c4
